app/src/lib/git/stash.ts

```diff
@@ -119,14 +120,7 @@ export async function createDesktopStashEntry(
 ): Promise<true> {
   // We must ensure that no untracked files are present before stashing
   // See https://github.com/desktop/desktop/pull/8085
-  if (untrackedFilesToStage.length > 0) {
-    const untrackedFilepaths = untrackedFilesToStage.map(uf => uf.path)
-    await git(
-      ['add', '--', ...untrackedFilepaths],
-      repository.path,
-      'stageUntrackedFilesBeforeStashing'
-    )
-  }
+  await stageFiles(repository, untrackedFilesToStage)
```

Comment:
I believe this will fail unless the user has explicitly checked the checkboxes for the untracked files. I think we need to tell `stageFiles` that we want to stage all contents of the untracked files.

```suggestion
  await stageFiles(
    repository,
    untrackedFilesToStage.map(x => x.withIncludeAll(true))
  )
```

If we don't the file will be treated as a partial commit (which I believe is a bug in and of itself but that's a separate story) and GitHub will attempt to apply a zero-length patch.

![image](https://user-images.githubusercontent.com/634063/65879648-91a7cd80-e390-11e9-91e7-6d8386e4f63a.png)

![2019-09-30 14-40-19 2019-09-30 14_40_39](https://user-images.githubusercontent.com/634063/65879671-9b313580-e390-11e9-9382-1bdfbc79bcf1.gif)

Reply:
Oh my! Great catch @niik! Fixed in 85472ba and documented in 8f0073b 👍